test/integration/project-page.test.js

```diff
@@ -117,14 +120,14 @@ describe('www-integration project-page signed in', () => {
     beforeAll(async () => {
         // expect(projectUrl).toBe(defined);
         driver = await buildDriver('www-integration project-page signed in');
-        await driver.get(rootUrl);
-        await driver.sleep(1000);
-        await signIn(username, password);
-        await findByXpath('//span[contains(@class, "profile-name")]');
     });
 
     beforeEach(async () => {
-        await driver.get(rootUrl);
+        // The browser may or may not retain cookies between tests, depending on configuration.
+        await navigate(rootUrl);
+        if (!await isSignedIn()) {
+            await signIn(username, password);
+        }
```

Comment:
Instead of just signing in once at the beginning, sign in _if necessary_ before each test. This covers situations like:
- tests are running in multiple browser windows and the browser is configured to discard cookies aggressively
- a problem in one test causes a sign-out that could interfere with other tests